streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -307,14 +340,22 @@ public StreamsConfig(Map<?, ?> props) {
 
     public Map<String, Object> getProducerConfigs(String clientId) {
         // generate producer configs from original properties and overridden maps
-        Map<String, Object> props = clientProps(ProducerConfig.configNames(), this.originals(), PRODUCER_DEFAULT_OVERRIDES);
-
+        Map<String, Object> props = clientProps(ProducerConfig.configNames(), getClientPropsWithPrefix(PRODUCER_PREFIX), PRODUCER_DEFAULT_OVERRIDES);
+        props.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, this.originals().get(BOOTSTRAP_SERVERS_CONFIG));
         // add client id with stream client id prefix
         props.put(CommonClientConfigs.CLIENT_ID_CONFIG, clientId + "-producer");
 
         return props;
     }
 
+    private Map<String, Object> getClientPropsWithPrefix(final String prefix) {
+        // to be backward compatible we first get all the props and then
+        // override them with the prefixed props
```

Comment:
nit: we are not "overriding" them, but duplicate them with the prefixed props right? If the user has already applied the prefix then this function would mostly be a no-op.